app/src/main/kotlin/com/wire/android/mapper/MessageContentMapper.kt

```diff
@@ -83,46 +83,46 @@ class MessageContentMapper @Inject constructor(
         }
     }
 
-    private suspend fun mapClientMessage(
-        message: Message.Client,
+    private suspend fun mapRegularMessage(
+        message: Message.Regular,
     ) = when (val content = message.content) {
         is Asset -> toAsset(
             conversationId = message.conversationId,
             messageId = message.id,
             assetContent = content.value
         )
-        is Text -> toText(
-            content = content.value,
-            editStatus = message.editStatus
-        )
-        else -> toText(null, Message.EditStatus.NotEdited)
+        else -> toText(content, message.editStatus)
     }
 
-    private fun toText(
-        content: String?,
+    fun toText(
+        content: MessageContent,
         editStatus: Message.EditStatus
-    ) = when (editStatus) {
-        is Message.EditStatus.Edited -> {
-            MessageContent.EditedMessage(
-                messageBody = MessageBody(message = toTextMessageContent(content)),
-                editTimeStamp = isoFormatter.fromISO8601ToTimeFormat(utcISO = editStatus.lastTimeStamp)
+    ) = MessageBody(when (content) {
+            is MessageContent.Text -> UIText.DynamicString(content.value)
+            is MessageContent.Unknown -> UIText.StringResource(
+                messageResourceProvider.sentAMessageWithContent, content.typeName ?: content::javaClass.name
```

Comment:
On prod, with minify enabled `content::javaClass.name` will be something like `c.d.a.f`. Better to just use `Unknown` instead?

Ideally, we should ask design to cover this. We should do something like displaying a warning that either the sender or self user needs to update their app, because the messate type is either too old or too new for us to read properly, and updating apps should fix it.